.pre-commit-config.yaml

```diff
@@ -7,6 +7,20 @@ repos:
         name: Check clang-format version
         entry: python3 ./ci/check-clang-format-version.py
         language: system
+      - id: check-ruff-format-version
+        name: Check ruff-format version
+        entry: python3 ./ci/check-ruff-version.py
+        language: system
+      - id: ruff
+        name: ruff
+        entry: ruff check
+        types_or: [python]
+        language: system
+      - id: ruff-format
+        name: ruff-format
+        entry: ruff format --check
+        types_or: [python]
+        language: system
```

Comment:
Shouldn't we use [`ruff-pre-commit`](https://github.com/astral-sh/ruff-pre-commit)?

It can make it easier to upgrade `ruff` if we need to

Reply:
Originally I think it's more flexible to do it on our side, but I think it also works to use other's scripts.